meson.build

```diff
@@ -44,6 +44,11 @@ else
     meson.add_dist_script(py, versioneer, '-o', '_version_meson.py')
 endif
 
+cy = meson.get_compiler('cython')
+if cy.version().version_compare('>=3.1.0')
+  add_global_arguments('-Xfreethreading_compatible=true', language : 'cython')
```

Comment:
I'd recommend `add_project_arguments` here instead of `add_global_arguments`, that's almost always the better choice (the only difference is that if someone uses pandas as a meson subproject, the `project` version doesn't affect the parent project).

Reply:
Thanks for the suggestion @rgommers! Updated.